### PR TITLE
Fix classifier

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
- [x] the `classifiers` will be an optional parameter with default value `["qubit_fit"]`
- [x] the `save_dir` will be optional, if absent the model will not be dumped
- [x] update the `ActionParser` in order to deal with optional parameters
- [x] fix dependencies

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Review by andrea-pasquale — APPROVED** [review]

File: src/qibocal/calibrations/characterization/calibrate_qubit_states.py [hunk 1]
```diff
@@ -69,11 +75,19 @@ def calibrate_qubit_states(
...
-    state0_results = platform.execute_pulse_sequence(state0_sequence, nshots=nshots)
+    state0_results = platform.execute_pulse_sequence(
+        state0_sequence,
+        ExecutionParameters(
+            nshots=nshots,
+            relaxation_time=relaxation_time,
+            acquisition_type=AcquisitionType.INTEGRATION,
+        ),
+    )
+    # platform.execute_pulse_sequence(state0_sequence, nshots=nshots)
```
Comment (AleCandido) [inline src/qibocal/calibrations/characterization/calibrate_qubit_states.py:86]: Dead code

File: src/qibocal/calibrations/characterization/calibrate_qubit_states.py [hunk 2]
```diff
@@ -84,11 +98,18 @@ def calibrate_qubit_states(
         data.add_data_from_dict(r)
 
     # execute the second pulse sequence
-    state1_results = platform.execute_pulse_sequence(state1_sequence, nshots=nshots)
-
+    # state1_results = platform.execute_pulse_sequence(state1_sequence, nshots=nshots)
```
Comment (AleCandido) [inline src/qibocal/calibrations/characterization/calibrate_qubit_states.py:101]: Dead code

File: src/qibocal/fitting/classifier/run.py [hunk 3]
```diff
@@ -91,15 +91,18 @@ def fit(self):
...
+        return False
 
     @property
     def hyperfile(self):
         r"""The path where the hyperparameters are stored."""
         return self.savedir / HYPERFILE
 
     def dump(self, path: pathlib.Path):
-        return self.mod.dump(self.trainable_model, path)
+        if base_dir:
```
Comment (AleCandido) [inline src/qibocal/fitting/classifier/run.py:104]: Here `base_dir` is not even in scope...

File: src/qibocal/fitting/classifier/run.py [hunk 4]
```diff
@@ -244,8 +250,10 @@ def train_qubit(
     qubit_data = qubits_data[qubits_data["qubit"] == qubit.name]
     nn_epochs = 200
     nn_val_split = 0.2
-    qubit_dir = base_dir / f"qubit{qubit.name}"
-    qubit_dir.mkdir(parents=True, exist_ok=True)
+    qubit_dir = None
+    if base_dir:
```
Comment (AleCandido) [inline src/qibocal/fitting/classifier/run.py:254]: ...

File: src/qibocal/fitting/classifier/run.py [hunk 5]
```diff
@@ -258,7 +266,8 @@ def train_qubit(
 
     for mod in classifiers:
         classifier = Classifier(mod, qubit_dir)
-        classifier.savedir.mkdir(exist_ok=True)
+        if classifier.savedir:
```
Comment (AleCandido) [inline src/qibocal/fitting/classifier/run.py:269]: What is the use case of not having a `savedir`?

File: src/qibocal/fitting/classifier/run.py [hunk 6]
```diff
@@ -289,12 +298,16 @@ def train_qubit(
...
-        dump_preds(y_pred, classifier.savedir)
-        classifier.dump(classifier.savedir / classifier.name)
+        if base_dir:
+            dump_preds(y_pred, classifier.savedir)
+            classifier.dump(classifier.savedir / classifier.name)
 
     benchmarks_table = pd.DataFrame([asdict(res) for res in results_list])
-    plots.plot_models_results(x_train, x_test, y_test, qubit_dir, models, names)
-    plots.plot_roc_curves(x_test, y_test, qubit_dir, models, names)
+    if base_dir:
```
Comment (AleCandido) [inline src/qibocal/fitting/classifier/run.py:306]: ...

**Comment by rodolfocarobene** [comment]
@Edoardo-Pedicillo I am now obsessed with documentation :-)

Could you maybe add something about the classifiers in the getting-started/examples? 
Or maybe just as a comment in calibrate_qubit_states... The fact is that is not easy to find which classifiers are implemented and the [documentation](https://qibo.science/qibocal/latest/api-reference/qibocal.html#classifiers) appears empty

**Review by andrea-pasquale — COMMENTED** [review]
LGTM.
I agree with @rodolfocarobene about the classifiers.
I think that for time being a detailed docstring should be enough.
The documentation is outdated since it still contain the old routines... I will open an issue to remember to update it.

**Review by rodolfocarobene — APPROVED** [review]

File: src/qibocal/calibrations/characterization/calibrate_qubit_states.py [hunk 7]
```diff
@@ -33,7 +33,18 @@ def calibrate_qubit_states(
         platform (:class:`qibolab.platforms.abstract.Platform`): custom abstract platform on which we perform the calibration.
         qubits (dict): Dict of target Qubit objects to perform the action
         nshots (int): number of times the pulse sequence will be repeated.
-        classifiers (list): list of classifiers
+        classifiers (list): list of classifiers, the available once are:
```
Comment (andrea-pasquale) [inline src/qibocal/calibrations/characterization/calibrate_qubit_states.py:36]:
```suggestion
        classifiers (list): list of classifiers, the available ones are:
```

**Pull request merged**